Mani8177/route task sample

manishk3189 opened the pull request:
Hi @doneill, I have create a sample Find route to address [#1145](https://devtopia.esri.com/runtime/java-android-sdk/issues/1145). Can you please review the same and let me know of changes if required?

Thanks

doneill commented:
@manishk3189 Please resolve conflicts with this branch before I start to review.

manishk3189 commented:
@doneill The conflicts are resolved now.

doneill commented on find-route/src/main/res/values-w820dp/dimens.xml:
```diff
@@ -0,0 +1,7 @@
+    <!-- Example customization of dimensions originally defined in res/values/dimens.xml
+         (such as screen margins) for screens with more than 820dp of available width. This
+         would include 7" and 10" devices in landscape (~960dp and ~1280dp respectively). -->
+    <dimen name="activity_horizontal_margin">64dp</dimen>
```
This resource appears to be unused.

doneill commented on find-route/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,6 @@
+<resources>
+    <!-- Default screen margins, per the Android Design guidelines. -->
+    <dimen name="activity_horizontal_margin">16dp</dimen>
+    <dimen name="activity_vertical_margin">16dp</dimen>
```
Both resources above appear to be unused.

doneill commented on find-route/src/main/java/com/esri/arcgisruntime/sample/findroute/MainActivity.java:
```diff
@@ -0,0 +1,292 @@
+
+    private static final String TAG = "FindRouteSample";
+    ArcGISVectorTiledLayer mVectorTiledLayer;
+    FloatingActionButton mDirectionFab;
```
Both field above should be marked as private.

doneill commented on find-route/src/main/java/com/esri/arcgisruntime/sample/findroute/MainActivity.java:
```diff
@@ -0,0 +1,292 @@
+                mProgressDialog.show();
+
+                getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+                getSupportActionBar().setHomeButtonEnabled(true);
```
`getSupportActionBar)` can be null, you should check for null before calling methods.

doneill commented:
@manishk3189 I provided some comments for your review.  From a UX perspective, I feel like the directions should interact with the map, e.g. tap on a direction step closes the drawer and highlights/zooms to that step on the map.  I don't think iOS or QT implemented that so I will follow up as I believe we should.

manishk3189 commented:
@doneill I have refactored the sample with the review comments and added zoom/highlight when clicked on direction maneuver from the list. I have also added `setViewpointWithDurationAsync` for better UX experience.

doneill commented on find-route/src/main/AndroidManifest.xml:
```diff
@@ -0,0 +1,22 @@
+        android:label="@string/app_name"
+        android:supportsRtl="true"
+        android:theme="@style/AppTheme">
+        <activity android:name="com.esri.arcgisruntime.sample.findroute.MainActivity">
```
Add the following `android:configChanges="orientation|screenSize|uiMode">` to ensure the activity does not restart when screen changes.

doneill commented:
@manishk3189 The sample is much improved, nice work.  I only have a couple comments left for you, the one above regarding screen orientation and the icons.  

The start and end icons are not pointing to the start/end positions.  You can see this when zoomed in from the directions list:  

Here is the start position: 
![start](https://cloud.githubusercontent.com/assets/1308001/16884292/71f91fca-4a7d-11e6-860a-c940f5469d98.png)

And the end position, which is a bit better but not exact. 
![end](https://cloud.githubusercontent.com/assets/1308001/16884317/8b9b2676-4a7d-11e6-9355-97f08ca35273.png)

Can you try and get these better positioned?

manishk3189 commented:
Hi @doneill I have pushed changes to address configuration changes and offsets to `PictureMarkerSymbols`. 
The screen privew is much better now.. 
![screen shot 2016-07-15 at 1 43 10 pm](https://cloud.githubusercontent.com/assets/12448081/16889945/2401a356-4a9f-11e6-9f48-4be054a73051.png)

--
![screen shot 2016-07-15 at 1 43 30 pm](https://cloud.githubusercontent.com/assets/12448081/16889952/3307af8a-4a9f-11e6-9836-0dffc73f02dd.png)

doneill commented:
@manishk3189 Looks good, will merge.
